spec/lib/subjects/seen_remover_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+RSpec.describe Subjects::SeenRemover do
+  let(:append_sms_ids) { (1..10).to_a }
+  let(:workflow) { instance_double("Workflow")}
+  let(:user) { instance_double("User") }
+  let(:uss) { instance_double("UserSeenSubject") }
+  let(:sms) { create(:set_member_subject) }
+  let(:subject_ids) { [ sms.subject.id ] }
+  let(:seens) do
+    build(:user_seen_subject,
+      user: nil,
+      workflow: nil,
+      subject_ids: subject_ids
+    )
+  end
+  subject { described_class.new(user, workflow, append_sms_ids) }
+
+  describe "#unseen_ids" do
+    it "should return the whole set if there are no seen subjects" do
+      allow(subject)
```

Comment:
I've never really liked stubbing of private methods on the object under test. Any way we could instead only stub interactions between objects, e.g. perhaps most of the work done by `seen_before_sms_ids` could be extracted to a class method/scope on SetMemberSubject? That way you can also easily test the behaviour of that (and we could decouple this service class from the DB entirely).